app/services/hmpps_api/prison_api/offender_api.rb

```diff
@@ -29,32 +29,43 @@ def self.list(prison, page = 0, page_size: 20)
 
+        complexities = HmppsApi::ComplexityApi.get_complexities offender_nos
         offenders = data.map do |payload|
           offender_no = payload.fetch('offenderNo')
-          HmppsApi::OffenderSummary.from_json(payload,
-                                              search_data.fetch(offender_no, {}),
-                                              latest_temp_movement: temp_movements[offender_no]).tap { |offender|
+          HmppsApi::OffenderSummary.new(payload,
+                                        search_data.fetch(offender_no, {}),
+                                        latest_temp_movement: temp_movements[offender_no],
+                                        complexity_level: complexities[offender_no]).tap { |offender|
             sentencing = sentence_details[offender.booking_id]
             offender.sentence = sentencing if sentencing.present?
           }
         end
         ApiPaginatedResponse.new(total_pages, offenders)
       end
 
-      def self.get_offender(offender_no)
+      def self.get_offender(raw_offender_no)
         # Bad NOMIS numbers mustn't produce invalid URLs
-        url_offender_no = URI.encode_www_form_component(offender_no)
-        route = "/prisoners/#{url_offender_no}"
-        response = client.get(route)
+        offender_no = URI.encode_www_form_component(raw_offender_no)
+        route = "/prisoners/#{offender_no}"
+        api_response = client.get(route).first
 
-        if response.empty?
+        if api_response.nil?
           nil
         else
-          search_data = get_search_payload([url_offender_no])
-          temp_movements = HmppsApi::PrisonApi::MovementApi.latest_temp_movement_for([url_offender_no])
-          HmppsApi::Offender.from_json(response.first,
-                                       search_data.fetch(url_offender_no, {}),
-                                       latest_temp_movement: temp_movements[url_offender_no]).tap do |offender|
+          search_data = get_search_payload([offender_no])
+          temp_movements = HmppsApi::PrisonApi::MovementApi.latest_temp_movement_for([offender_no])
+          prisoner = if PrisonService::womens_prison?(api_response.fetch('latestLocationId'))
+                       HmppsApi::Offender.new api_response,
+                                              search_data.fetch(offender_no, {}),
+                                              latest_temp_movement: temp_movements[offender_no],
+                                              complexity_level: HmppsApi::ComplexityApi.get_complexity(offender_no)
+                     else
+                       HmppsApi::Offender.new api_response,
+                                              search_data.fetch(offender_no, {}),
+                                              latest_temp_movement: temp_movements[offender_no],
+                                              complexity_level: nil
+                     end
```

Comment:
We could DRY this up by only making `complexity_level` conditional – since that's the only thing that changes between the two branches of `prisoner`.

IMHO this is a little easier to read/follow because it's obvious what changes and what stays the same.

```suggestion
          complexity_level = if PrisonService::womens_prison?(api_response.fetch('latestLocationId'))
                               HmppsApi::ComplexityApi.get_complexity(offender_no)
                             end
          prisoner = HmppsApi::Offender.new api_response,
                                            search_data.fetch(offender_no, {}),
                                            latest_temp_movement: temp_movements[offender_no],
                                            complexity_level: complexity_level
```